javascript/javascript1/week2/flight_booking_formal.js

```diff
@@ -0,0 +1,36 @@
+function getFullname(firstName, secondName, useFormalName) {
+    useFormalName === true;
+    let fullname = `${firstName} ${secondName}`;
+    if (useFormalName) {
+        return `Lord ${fullname}`
+    } else if(useFormalName === false || useFormalName === undefined) {
+        return fullname;
+    }
+}
+
+console.log(getFullname("John", "Doe", true, "Male"))
+
+// Fix for women //
+function getFullnameWithGender(firstName, secondName, useFormalName, gender) {
+    if (firstName=== undefined || secondName=== undefined) {
+      return "Please specify your name"
+    }
+      useFormalName = true
+      formal ="";
+      if (gender === undefined) {
+          formal = undefined
+          useFormalName = false
+        }else if (gender === "Male") {
+          formal = "Lord"
+          }else if (gender = "Female") {
+          formal = "Lady"
+          }
+      let fullname = `${firstName} ${secondName}`;
+      if (useFormalName) {
+          return `${formal} ${fullname}`
+      } else if(useFormalName === false || useFormalName === undefined) {
+          return fullname;
+      }
+  }
```

Comment:
- You're comparing against `undefined` but it currently doesn't fail with empty strings.
- Also if the gender is empty it assumes one.

```js
> getFullnameWithGender("", "", true, "Male")
'Lord  '

> getFullnameWithGender("Marco", "Rich", true, "")
'Lady Marco Richetta'
```